### PR TITLE
modified login command to include host key

### PR DESCRIPTION
**Pull request by facchettos**
### Which issue this PR addresses:
now includes the ssh host public key for easier login (saw -> portal leg)

Fixes
https://issues.redhat.com/browse/ARO-3024 
### Test plan for issue:
tests are still present

### CODE REVIEW

File: pkg/portal/ssh/ssh_test.go [hunk 1]
```diff
@@ -60,7 +66,11 @@ func TestNew(t *testing.T) {
 				})
 			},
 			wantStatusCode: http.StatusOK,
-			wantBody:       "{\n    \"command\": \"ssh username@localhost\",\n    \"password\": \"03030303-0303-0303-0303-030303030001\"\n}",
+			wantBody: `{
+    "command": "echo 'localhost ssh-rsa ` + hostPubFingerprint + `' > localhost_known_host ; ssh -o UserKnownHostsFile=localhost_know_host username@localhost",
```
Comment (dem4gus) [inline pkg/portal/ssh/ssh_test.go:70]: Same typo

```suggestion
    "command": "echo 'localhost ssh-rsa ` + hostPubFingerprint + `' > localhost_known_host ; ssh -o UserKnownHostsFile=localhost_known_host username@localhost",
```

**Review by dem4gus — CHANGES_REQUESTED** [review]
Just some small typos.

File: pkg/portal/ssh/ssh.go [hunk 2]
```diff
@@ -159,29 +168,57 @@ func (s *SSH) New(w http.ResponseWriter, r *http.Request) {
...
 }
 
 func (s *SSH) internalServerError(w http.ResponseWriter, err error) {
 	s.log.Warn(err)
 	http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 }
+
+const (
+	sshCommand = "echo '{{ .Hostname }} ssh-rsa {{.PublicKeyFingerprint}}' > {{.Hostname}}_known_host ; " +
+		"ssh -o UserKnownHostsFile={{.Hostname}}_know_host{{if .IsLocalDevelopmentMode}} -p 2222{{end}} {{.User}}@{{.Hostname}}"
```
[the remark this answers is not in the document]
Comment (facchettos) [reply pkg/portal/ssh/ssh.go]: good catch

**Review by dem4gus — APPROVED** [review]

File: pkg/portal/ssh/ssh.go [hunk 3]
```diff
@@ -159,29 +168,58 @@ func (s *SSH) New(w http.ResponseWriter, r *http.Request) {
...
 	s.log.Warn(err)
 	http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 }
+
+const (
+	sshCommand = "echo '{{ .Hostname }} ssh-rsa {{.PublicKeyFingerprint}}' > {{.Hostname}}_known_host ; " +
+		"ssh -o UserKnownHostsFile={{.Hostname}}_known_host{{if .IsLocalDevelopmentMode}} -p 2222{{end}} {{.User}}@{{.Hostname}}"
+)
+
+func createLoginCommand(IsLocalDevelopmentMode bool, user, host, fingerPrint string) (string, error) {
```
Comment (s-amann) [inline pkg/portal/ssh/ssh.go:210]: nit: IsLocalDevelopmentMode -> isLocalDevelopmentMode

File: pkg/portal/ssh/ssh.go [hunk 4]
```diff
@@ -159,29 +168,58 @@ func (s *SSH) New(w http.ResponseWriter, r *http.Request) {
...
+		return "", err
+	}
+	type fields struct {
+		User                   string
+		Hostname               string
+		PublicKeyFingerprint   string
+		IsLocalDevelopmentMode bool
+	}
+	var buff bytes.Buffer
+	err = tmp.Execute(&buff, fields{user, host, fingerPrint, IsLocalDevelopmentMode})
```
Comment (s-amann) [inline pkg/portal/ssh/ssh.go:223]: why do we call Execute during command creation?

**Review by s-amann — COMMENTED** [review]
1 nit and a question, otherwise lgtm

File: pkg/portal/ssh/ssh.go [hunk 5]
```diff
@@ -159,29 +168,58 @@ func (s *SSH) New(w http.ResponseWriter, r *http.Request) {
...
-	_, _ = w.Write(b)
 }
 
 func (s *SSH) internalServerError(w http.ResponseWriter, err error) {
 	s.log.Warn(err)
 	http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
 }
+
+const (
+	sshCommand = "echo '{{ .Hostname }} ssh-rsa {{.PublicKeyFingerprint}}' > {{.Hostname}}_known_host ; " +
```
Comment (SudoBrendan) [inline pkg/portal/ssh/ssh.go:206]: Do we ever clean up these files or would they stay on the RP until we redeploy?

Comment (facchettos) [reply pkg/portal/ssh/ssh.go:206]: this command is not run on the rp, nor the portal. This is executed on the saw

File: pkg/portal/ssh/ssh.go (hunk 4 above)
Comment (facchettos) [reply pkg/portal/ssh/ssh.go:223]: when do you want to call it? the username is unique to each request, so we cannot just run it once.

Comment (facchettos) [reply pkg/portal/ssh/ssh.go]: Oh I think I understand the confusion now, I'm not executing the command, I'm executing the text template

**Comment by facchettos** [comment]
I had to import the `knownhost` package, which in turn updated a few libraries in `golang.org/x/`

File: pkg/portal/ssh/ssh.go (hunk 5 above)
Comment (jhoreman) [reply pkg/portal/ssh/ssh.go]: Hey Jeremy, thanks for this. Can you elaborate on this comment? How would this change our current workflow of connecting?

Comment (facchettos) [reply pkg/portal/ssh/ssh.go]: This is just changing the command we have to copy paste from the saw

Comment (facchettos) [reply pkg/portal/ssh/ssh.go]: There is no change in the flow

Comment (jhoreman) [reply pkg/portal/ssh/ssh.go]: Ack, understand the flow now after review.

**Review by jhoreman — APPROVED** [review]

File: pkg/portal/ssh/ssh.go (hunk 4 above)
Comment (s-amann) [reply pkg/portal/ssh/ssh.go]: aye, thanks for clearing that up.

**Review by s-amann — APPROVED** [review]

**Review by dem4gus — APPROVED** [review]
LGTM

**Pull request merged**